fluent/fluent.go

```diff
@@ -525,24 +526,24 @@ func (n *nextHopEntry) WithNextHopNetworkInstance(ni string) *nextHopEntry {
 
 // TODO(robjs): add additional NextHopEntry fields.
 
-// opproto implements the gRIBIEntry interface, returning a gRIBI AFTOperation. ID
+// opproto implements the GRIBIEntry interface, building a gRIBI AFTOperation. ID
 // and ElectionID are explicitly not populated such that they can be populated by
 // the function (e.g., AddEntry) to which they are an argument.
 func (n *nextHopEntry) opproto() (*spb.AFTOperation, error) {
 	return &spb.AFTOperation{
 		NetworkInstance: n.ni,
 		Entry: &spb.AFTOperation_NextHop{
-			NextHop: n.pb,
+			NextHop: proto.Clone(n.pb).(*aftpb.Afts_NextHopKey),
 		},
 	}, nil
 }
 
-// entryproto implements the gRIBIEntry interface, returning a gRIBI AFTEntry.
+// entryproto implements the GRIBIEntry interface, building a gRIBI AFTEntry.
 func (n *nextHopEntry) entryproto() (*spb.AFTEntry, error) {
 	return &spb.AFTEntry{
 		NetworkInstance: n.ni,
 		Entry: &spb.AFTEntry_NextHop{
-			NextHop: n.pb,
+			NextHop: proto.Clone(n.pb).(*aftpb.Afts_NextHopKey),
```

Comment:
This looks fine per our discussion. It is safe to clone given that you are now exposing the entry that can be shared with something else outside this library.